mlflow/utils/_spark_utils.py

```diff
@@ -150,3 +151,27 @@ def get_or_extract(archive_path):
 
         _SparkDirectoryDistributor._extracted_dir_paths[archive_path] = temp_dir
         return _SparkDirectoryDistributor._extracted_dir_paths[archive_path]
+
+
+@contextlib.contextmanager
+def modified_environ(update):
+    """
+    Temporarily updates the ``os.environ`` dictionary in-place.
+
+    The ``os.environ`` dictionary is updated in-place so that the modification
+    is sure to work in all situations.
+
+    :param update: Dictionary of environment variables and values to add/update.
+    """
+    update = update or {}
+    original_env = {k: os.environ.get(k) for k in update}
+
+    try:
+        os.environ.update(update)
+        yield
+    finally:
+        for k, v in original_env.items():
+            if v is None:
+                os.environ.pop(k)
```

Comment:
```suggestion
                os.environ.pop(k, None)
```

because there is no guarantee that `os.environ` still has `k`.

Reply:
good catch @harupy